### DISCUSSION TITLE
Water Background shows up at the very top and it's inverted

### DISCUSSION THREAD

**Discussion by RunninPigeon** [TrianglyRU/OrbinautFramework#28] [Q&A] [answered]
im trying to get the water background to the water height of 1240 but it keeps showing up at the top and inverted for some reason:
`
			var _bg_dist = dist_get_data(EFFECTDATA.LZBG);
			var _fg_dist = dist_get_data(EFFECTDATA.LZFG);
			
			dist_set_fg([], _fg_dist, -0.5, 0, room_height, ["GraphicsA", "GraphicsB"]);
			dist_set_bg(_bg_dist, _bg_dist, -0.25, 192, 348);
			
			bg_set_colour(#C0C0C0);
			bg_add_layer(spr_trz_bg_layer0, 0, 104, 192, 0, 0, 0, 0.300, 0);			// Water Layer
			bg_set_perspective_x(1.0, 2);
			bg_set_perspective_y(water_level_init);`
![Orbinaut Framework 2 9_7_2024 3_28_15 PM](https://github.com/user-attachments/assets/c13ce744-9b1d-4fc7-a71d-e73e46a5b73b)

**Comment by TrianglyRU** [comment]
`_water_pos` argument in `m_stage_set` is not set to 1240 if that's the case

**Reply by RunninPigeon** [reply]
i have `m_stage_set(1, "TESTING ROOM", ACT_SINGLE, bgm_woofle, [spr_obj_animal_flicky, spr_obj_animal_pocky], room_height, 1240, rm_devmenu, false);`

**Reply by TrianglyRU** [reply]
![изображение](https://github.com/user-attachments/assets/e42ea976-39ed-432c-853e-5d5883f8cc58)
Please, export your project and attach it here (upload somewhere else if it is too big). I'll take a look

**Reply by RunninPigeon** [reply]
okay then https://www.mediafire.com/file/y1evo9dcjv2fh6r/OrbinautFramework2.yyz/file

**Reply by TrianglyRU** [reply, marked as answer]
`m_stage_set()` should be the very first function called
[Now, let's configure c_stage specifically for `rm_myroom`. Start by calling the `m_stage_set()` function."](https://github.com/TrianglyRU/OrbinautFramework/wiki/6.-Creating-a-New-Room-or-Stage#setting-up-the-stage)

Also, you have no idea how arguments work if you try to pass to a function an argument from another function. Please, spend some time to learns the basics of programming and GameMaker Language.

Your code should be like this:
```gml
case rm_stage_trz0:
		
    m_stage_set(1, "TESTING ROOM", ACT_SINGLE, bgm_woofle, [spr_obj_animal_flicky, spr_obj_animal_pocky], room_height, 1240, rm_devmenu, false);    
			
    bg_set_colour(#C0C0C0);
    bg_add_layer(spr_trz_bg_layer0, 0, 104, 192, 0, 0, 0, 0.300, 0);			// Water Layer
    bg_set_perspective_x(1.0, 2);
    bg_set_perspective_y(water_level_init);	
    
    player_spawn(80, 1200, global.player_main, "Objects");
    player_spawn(64, 1200, global.player_cpu, "Objects");
    collision_load_binary("widths_s1", "heights_s1", "angles_s1", "CollisionA", "CollisionB");
        
break;
```

Also, make sure to not make the background 3000px wide if you don't want your device struggle trying to run a 2d game.

Answering your next question:
The water background won't oscillate with the water level. Open `Step Event` of `c_stage` and find this code:
```gml
// Adjust water level
if water_enabled
{
	water_level = obj_oscillate_y(water_level_init, _c_framework.frame_counter * ANGLE_INCREMENT, 10, 1, 90);
	
	if room == rm_stage_tsz0
	{
		c_framework.background.perspective_y_data[0] = water_level;
	}
}
```

Here for your room you either call `c_framework.background.perspective_y_data[0] = water_level;` as well or do not oscilllate the `water_level`  at all

**Reply by RunninPigeon** [reply]
thank you! C:
